dlock/dlock.go

```diff
@@ -132,20 +133,25 @@ func New(cfg *LockConfig) (func(), error) {
 		})
 	}()
 
-	<-lockAcquired
+	select {
+	case <-lockAcquired:
+		logger.Debugw("Lock acquired")
 
-	logger.Debugw("Lock acquired")
+		unlock := func() {
+			cancel()
 
-	unlock := func() {
-		cancel()
+			// Wait for `leaderelection.RunOrDie` to terminate. Otherwise
+			// the main thread my already terminate the process before the
+			// lock is released.
+			wg.Wait()
 
-		// Wait for `leaderelection.RunOrDie` to terminate. Otherwise
-		// the main thread my already terminate the process before the
-		// lock is released.
-		wg.Wait()
+			logger.Infow("Lock released")
+		}
+
+		return unlock, nil
 
-		logger.Infow("Lock released")
+	case <-ctx.Done():
```

Comment:
Just to be sure: if the external context is cancelled, there is no way that this function can return no error, right?

Reply:
No, if the context is cancelled we just stop `RunOrDie`, then we will block on <-lockacquired (that's why I use now a `select` statement)